src/services/HTTPTransport.ts

```diff
@@ -0,0 +1,92 @@
+export enum METHODS {
+  GET = "GET",
+  PUT = "PUT",
+  POST = "POST",
+  DELETE = "DELETE"
+}
+
+interface Options {
+  method: string;
+  headers: Record<string, string>;
+  data: string;
+  timeout: number;
+}
+
+function queryStringify(data: any): string {
+  let outcome = "?";
+  Object.keys(data).forEach((key, i, arr) => {
+    outcome += `${key}=${data[key]}`;
+    if (arr.length > 1 && i !== arr.length - 1) {
+      outcome += `&`;
+    }
+  });
+
+  return outcome;
```

Comment:
выглядит довольно сложно, я бы упростил
```suggestion
  return "?" + Object
 .entries(data)
 .map(([key, value]) => `${key}=${value}`)
 .join("&")
```